tb_plugin/torch_tb_profiler/profiler/overall_parser.py

```diff
@@ -299,15 +309,54 @@ def parse_events(self, events, runtime_node_list, device_node_list):
             for cost_index in range(len(self.avg_costs.costs)):
                 self.avg_costs.costs[cost_index] += self.steps_costs[i].costs[cost_index]
 
+            comm_costs = OverallParser.StepCommunicationCosts()
+            comm_costs.overlap = get_ranges_sum(intersection_ranges_lists([self.steps[i]], comm_kernel_overlap))
+            comm_costs.computation = get_ranges_sum(intersection_ranges_lists([self.steps[i]], self.role_ranges[ProfileRole.Kernel]))
+            comm_costs.communication = get_ranges_sum(intersection_ranges_lists([self.steps[i]], self.role_ranges[ProfileRole.Communication]))
+            comm_costs.other = self.steps_costs[i].costs[ProfileRole.Total] + comm_costs.overlap - comm_costs.computation - comm_costs.communication
+            self.communication_overlap.append(comm_costs)
+
         for i in range(len(self.avg_costs.costs)):
             self.avg_costs.costs[i] /= valid_steps
 
-    def parse_event(self, event):
+        # Sort the communication node according the start time, this is for correlating communication node between workers
+        for comm_node in context_data.communication_data.values():
+            comm_node.kernel_ranges.sort(key=lambda x: (x[0], -x[1]))
+            self.comm_node_list.append(comm_node)
+        self.comm_node_list.sort(key=lambda x: (x.start_time, -x.end_time))
+
+        # Find each communication node belong to which step
+        index = 0
+        for comm_node in self.comm_node_list:
+            while index < valid_steps:
+                if comm_node.start_time >= self.steps[index][0] and comm_node.end_time <= self.steps[index][1]:
+                    comm_node.step_name = self.steps_names[index]
+                    break
+                elif comm_node.start_time >= self.steps[index][1]:
+                    index += 1
+                else:
+                    logger.error("Found a communication op not belong to any step.")
+                    break
+            if index >= valid_steps:
+                logger.error("Found communication ops not belong to any step. ")
+                break
+
+    def parse_event(self, event, communication_data):
         ts = event.ts
         dur = event.duration
         evt_type = event.type
+        external_id = None
+        if "external id" in event.args:
```

Comment:
better to extract the external id logic to TraceEvent class so that both module_parse.py and here can reuse the logic

Reply:
Yes, thanks, done